src/main/java/com/atlassian/db/replica/internal/ReplicaPreparedStatement.java

```diff
@@ -78,22 +67,33 @@ protected ReplicaPreparedStatement(
     @Override
     public ResultSet executeQuery() throws SQLException {
         checkClosed();
-        final PreparedStatement statement = getReadStatement(sql);
-        return execute(statement::executeQuery);
+        final QueryContext.Builder contextBuilder = QueryContext.builder(Reason.READ_OPERATION, false).sql(sql);
```

Comment:
> maybe getReadStatement could hold that default?

Yes. I think getReadStatement could hold the default.

> If not, then at least it could read the sql from context, because now it gets it via both params.

Context#Builder is an API, and the SQL was needed only for internal usage here.

Reply:
Maybe Context#Builder  shouldn't be part of API 🤔

Reply:
if we don't have a concrete case for new API, we should hide it